File: 프로그래머스/unrated/181830. 정사각형으로 만들기/정사각형으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr) {
    int col = arr.size();
    int row = arr[0].size();
    
    if (arr.size() < arr[0].size())
        col = arr[0].size();
    else if (arr.size() > arr[0].size())
        row = arr.size();
    else
        return arr;
    
    vector<vector<int>> answer(col, vector<int> (row, 0));
    
    for (int i = 0; i < arr.size(); i++)
        for (int j = 0; j < arr[0].size(); j++)
            answer[i][j] = arr[i][j];
    
    return answer;
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.java
class Solution {
    public int[][] solution(int[][] arr) {
        int max = Integer.max(arr.length, arr[0].length);
        int[][] newArr = new int[max][max];
        int idx = 0;
        for (int[] arrs : arr) {
            System.arraycopy(arrs, 0, newArr[idx], 0, arrs.length);
            idx++;
        }

        return newArr;
    }
}